doc/languages-frameworks/beam.section.md

```diff
@@ -56,11 +56,65 @@ Mix functions similarly to Rebar3, except we use `buildMix` instead of `buildReb
 
 Alternatively, we can use `buildHex` as a shortcut:
 
+#### buildMix - Elixir Phoenix example
+
+Here is how your default.nix file would look like
+
+```default.nix
+with import <nixpkgs-unstable> { };
+
+let
+  packages = beam.packagesWith beam.interpreters.erlang;
+  src = builtins.fetchGit {
+    url = "ssh://git@github.com/your_id/your_repo";
+    rev = "replace_with_your_commit";
+  };
+
+  nodeDependencies =
+    (pkgs.callPackage ./assets/default.nix { }).shell.nodeDependencies;
+
+in packages.buildMix {
+  inherit src;
+  name = "your_project";
+  version = "0.0.1";
+  mixEnv = "prod";
+  # leave this empty and nix will complain and tell you the right value
+  # to replace this with
+  depsSha256 = "";
+  inherit src;
+  depsPreConfigure = ''
+    # if you have build time environment variable add them here
+    export MY_ENV_VAR="my_value";
+  '';
+  preConfigure = ''
+    # if you have build time environment variable add them here
+    export MY_ENV_VAR="my_value";
```

Comment:
Is this the same as the one in the section above? So build environment variables must be exported in both steps? You can make them available to all steps by sending them to the underlying mkDerivation as Nix variables such as `MY_ENV_VAR = "my_value";`. You could add an attribute set that you include in the args sent to mkDerivation to make it clearer.

Reply:
You're right. Adding an attribute called `build_env_vars = { MY_ENV_VAR = "my_value";};` then using a merge in the mkDerivation in both the build and the fetch steps to make them available.
Thanks!

Reply:
I added a buildEnvVars attribute. Trying to merge this and the default attrs resulted in an error, so I have removed the merging of the attrs with the rest of the variables for mkDerivation. What changes is that you can't override as easily steps for the derivartion. Here in particular, the preConfigure hook is not overridable anymore from passed attrs. It looses a bit of flexibility. Not sure how to proceed otherwise.